### PR TITLE
MemVerge MMCloud Agent

### PR DESCRIPTION
**Pull request by edwinyyyu**
# TL;DR
This plugin allows executing Flyte tasks using MemVerge Memory Machine Cloud.

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [x] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Critical correctness dependencies:
* https://github.com/flyteorg/flyteplugins/pull/391
* https://github.com/flyteorg/flyteplugins/pull/395

Job submission performance dependency:
* https://github.com/flyteorg/flytepropeller/pull/590

Relevant bug fixes:
* https://github.com/flyteorg/flyteplugins/pull/392

### CODE REVIEW

File: plugins/flytekit-mmcloud/setup.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    namespace_packages=["flytekitplugins"],
+    packages=[f"flytekitplugins.{PLUGIN_NAME}"],
+    install_requires=plugin_requires,
+    license="apache2",
+    python_requires=">=3.9",
+    classifiers=[
+        "Intended Audience :: Science/Research",
+        "Intended Audience :: Developers",
+        "License :: OSI Approved :: Apache Software License",
+        "Programming Language :: Python :: 3.9",
```
[the remark this answers is not in the document]
Comment (edwinyyyu) [reply plugins/flytekit-mmcloud/setup.py:28]: I've used https://peps.python.org/pep-0585/ new type annotations. Should I change these?

File: plugins/flytekit-mmcloud/flytekitplugins/mmcloud/agent.py [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+        Synchronous get() for compatibility with older flytekit versions.
+        """
+        return asyncio.run(self.async_get(context=context, resource_meta=resource_meta))
+
+    def delete(self, context: grpc.ServicerContext, resource_meta: bytes) -> DeleteTaskResponse:
+        """
+        Synchronous delete() for compatibility with older flytekit versions.
+        """
+        return asyncio.run(self.async_delete(context=context, resource_meta=resource_meta))
+
```
[the remark this answers is not in the document]
Comment (edwinyyyu) [reply plugins/flytekit-mmcloud/flytekitplugins/mmcloud/agent.py:242]: Is it not helpful for compatibility with older flytekit versions? I can remove it if it's better not to have it.

File: plugins/flytekit-mmcloud/setup.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (pingsutw) [reply plugins/flytekit-mmcloud/setup.py:28]: yes, use `typing.Dict`, then we will be able to support python 3.8

**Review by pingsutw — DISMISSED** [review]

**Review by pingsutw — APPROVED** [review]

**Review by pingsutw — APPROVED** [review]
